### ISSUE TITLE
Enable collaborators to manage issues and pull requests

### ISSUE THREAD

**Issue by ajafff** [palantir/tslint#2985]
Recently this repository is lacking active mainainers.
Issues pile up, Pull Requests go stale and authors lose interest, important bug fixes don't get released in a reasonable amount of time because it's simply too much for one person to handle.

This is not to blame anyone. I know you don't do this as full time job and have other stuff to do. It also seems like @adidahiya is the only maintainer recently.
But hey, this is open source. There are some pretty active and talented contributors that would be willing to help.

Get collaborators involved in triaging issues, reviewing and merging pull requests. That works great for other projects like node or the whole levelup ecosystem. IMO this will increase productivity:
* decreased number of open issues makes it easier to find bugs to fix
* ability to assign someone to issues avoids unnecessary work of others
* earlier feedback on PRs keeps authors interested
* more eyes to find bugs
* faster merging of bug fixes
* increased activity attracts more contributors
* (I can go on vacation without this project going completely inactive)

Of course there's always some risk in giving away commit access. But there are some things to consider here:
* Write a "code of conduct" for a standard way to manage this repo.
* With an increasing number of collaborators there are more eyes reviewing pull requests.
* It's git after all and everything can be reverted or even rebased out of the history.
* You retain the rights to publish a release.

Some interesting reads on the topic:
[Contributing guidelines of level projects (OPEN open source)](https://github.com/Level/community/blob/master/CONTRIBUTING.md)
[How to be an open source gardener](http://words.steveklabnik.com/how-to-be-an-open-source-gardener)

Btw. I'm not proposing this because I need the fame. I really want to get this project forward. In fact I'm frustrated by the status quo and start losing my enthusiams.

**Comment by adidahiya** [comment]
I agree, I'd like to add more collaborators to this repo. I'll work on getting a code of conduct added here this week. Can you also send me an email? Address in my bio.

**Comment by adidahiya** [comment]
> You retain the rights to publish a release.

I was poking around at this and I don't see a way to restrict git tagging (to publish a release) to certain users. Do you know of a way to do this?

**Comment by ajafff** [comment]
> I don't see a way to restrict git tagging (to publish a release) to certain users. Do you know of a way to do this?

Initially I thought you just don't give away publish permissions for npm. But now I see that it's automatically published once the tag is added...
Unfortunately there seems to be no option to disable pushing tags to github.

An alternative might be to check CIRCLE_USERNAME in CI before actually publishing.

**Comment by IllusionMH** [comment]
> An alternative might be to check CIRCLE_USERNAME in CI before actually publishing.

Only if config will be stored in separate repo, otherwise it might be alternated.
Oher options:
* Task for publishing which can be invoked only manually
* Trust @ajafff and future collaborators with write access that they won't push tags/edit list of allowed Github/CI names

P.S. Nice to see that project is open for active collaborators! 🎆

**Comment by ajafff** [comment]
> Only if config will be stored in separate repo, otherwise it might be alternated.

You're right. I thought there was a the ability to add a command in the CircleCI settings that's executed before the deploy script is run.
Seems like it's only possible for `pre/post-dependency` and `pre/post-test` but nothing like that for deploy :disappointed:

**Comment by JKillian** [comment]
Hey @ajafff, super happy to hear you're interested in being a maintainer! I appreciate the work you've done on TSLint and your quality input on many TSLint-related discussions.

As far as the release process, I can't see any sort of CircleCI features that would allow us to do things securely. [There's a brief discussion here](https://discuss.circleci.com/t/secure-circle-yml-file/5277), but no good answer. Seems like for now our best bet is to add a manual release step - it might be good to have it build TSLint from scratch in a clean temporary directory before releasing and/or use [`package.json` `files`](https://docs.npmjs.com/files/package.json#files) to make sure we're not accidentally including extra files from a dev's local machine in our releases.

**Comment by JKillian** [comment]
@adidahiya, @nchen63 have either of you started doing the work to change the deployment process? If not, I can grab do it today or tomorrow. Would like to get this taken care of so we can add @ajafff (and others down the road) as collaborators!

**Comment by ajafff** [comment]
@adidahiya @JKillian I really enjoy being a collaborator. But the last 4 weeks I've been very lonely managing this repo.

If you are not available, there is nobody to review and merge my pull requests.
I can't even merge some simple bug fixes on my own. That's a good thing, because I like to get feedback on my code.
I'd also prefer a second opinion on pull request or feature requests of other contributors before merging.

Long story short: I want to invite you to add more collaborators.
I could name some suitable collaborators, but I don't want anyone to be disappointed if you disagree.
So how would you like to approach this topic? An intermittent public nomination thread or rather a private discussion/voting?

**Comment by ajafff** [comment]
Kindly bumping this issue. It's been a month since my previous comment and nothing has changed. Please don't leave this project unattended due to lack of time.

**Comment by JKillian** [comment]
@ajafff - sorry that nobody has responded to your comment! I concur with everything you said - having additional feedback and support from more collaborators would be a good thing (although your work has been fantastic @ajafff so thank you for that).

Please send a private email about who you think would be a suitable collaborator (no sense to make someone feel bad if we disagree on it) and if we all agree we can start getting them onboarded!

**Comment by johnwiseheart** [comment]
Hey all,

@suchanlee, @walkerburgin and I have recently come on as new maintainers. It's going to take us a little while to ramp up, but we've already started categorizing some issues, as well as looking into PRs. Its unfortunate that the repo went without active maintainers for some time, but we hope to improve the situation and bring the repo back to a healthy status. @ajafff - thank you for continuing on for many months without us - you've done a great job, and I hope we can being back your enthusiasm!

John & the TSLint team

**Comment by JKillian** [comment]
Wooh, welcome @johnwiseheart / @suchanlee  / @walkerburgin! Going to close out this issue as I believe it has been addressed by your joining. However, others are of course still welcome to get involved whether by making PRs, reviewing PRs, or other ways if interested.
